tests/app/celery/test_process_delivery_status_result_tasks.py

```diff
@@ -313,3 +318,46 @@ def test_wt_delivery_status_callback_should_log_total_time(
         NOTIFICATION_DELIVERED,
         'twilio',
     )
+
+
+@pytest.mark.serial
+def test_process_delivery_status_no_status_reason_for_delivered(
+    notify_db_session,
+    mocker,
+    sample_template,
+    sample_notification,
+    sample_delivery_status_result_message,
+    sample_notification_platform_status,
+):
+    """
+    When a notification is updated to "delivered" status, its "status_reason" should be set to
+    the empty string.
+    """
+
+    # This test is marked "serial" because the reference is used by many tests.  Making it a random
+    # value causes the test to fail.
+    notification = sample_notification(
+        template=sample_template(),
+        reference='SMhardcodedKWM',
+        sent_at=datetime.datetime.utcnow(),
+        status=NOTIFICATION_SENT,
+        status_reason='This is not the empty string.',
+    )
+    assert notification.reference == 'SMhardcodedKWM'
+    assert notification.status == NOTIFICATION_SENT
+    assert notification.status_reason
+
+    mocker.patch('app.celery.process_delivery_status_result_tasks._get_include_payload_status', returns=True)
+    mocker.patch(
+        'app.celery.process_delivery_status_result_tasks._get_notification_parameters',
+        return_value=tuple(sample_notification_platform_status.values()) + (1, 1),
+    )
+    callback_mock = mocker.patch('app.celery.process_delivery_status_result_tasks.check_and_queue_callback_task')
+
+    assert process_delivery_status(event=sample_delivery_status_result_message)
+    callback_mock.assert_called_once()
+
+    notify_db_session.session.refresh(notification)
+    assert notification.reference == 'SMhardcodedKWM'
+    assert notification.status == NOTIFICATION_DELIVERED
+    assert not notification.status_reason, 'This should be the empty string.'
```

Comment:
nit, not a required change.
You duplicate the reference and status reason a few times. Probably would have been best to create a data dictionary or use some setup variables at the top of the method.

Reply:
I didn't make any changes for this due to time constraints.